src/redisearch.h

```diff
@@ -102,18 +102,18 @@ typedef struct RSDocumentMetadata_s {
 
   /* Document flags  */
   RSDocumentFlags flags : 8;
+  // Type of source document. Hash or JSON.
+  DocumentType type : 8;
 
+  uint32_t ref_count : 16;
   /* Optional user payload */
   RSPayload *payload;
 
   struct RSSortingVector *sortVector;
   /* Offsets of all terms in the document (in bytes). Used by highlighter */
   struct RSByteOffsets *byteOffsets;
```

Comment:
@MeirShpilraien WDYT about allocating pointers for payload, sortVector, and byteOffsets dynamically?
Especially payload which is deprecated.
Going from 72 bytes to 64 might even help the system to optimize allocations since it is `pow(2)`.

Reply:
You mean putting it on another struct and have a pointer to it which might be NULL? If so you will have to allocate it if one of them exists. If you go with dynamically allocate the dmd size, you will need flag that tells you what fields it has. I am not sure how much you will spare. Am I missing something?